### PR TITLE
[syncer part 1] add multi cluster controller runtime and entry syncer…

### PR DESCRIPTION
**Pull request by adohe**
… command

### CODE REVIEW

File: incubator/virtualcluster/pkg/resource-syncer/controller/controller.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		klog.Warning("Work item is not a Request. Ignore it. Next.")
+		c.Queue.Forget(obj)
+		return true
+	}
+
+	if result, err := c.reconciler.Reconcile(req); err != nil {
+		klog.Error(err)
+		klog.Error("Could not reconcile Request. Stop working.")
+		c.Queue.AddRateLimited(req)
+		return false
```
Comment (Fei-Guo) [inline incubator/virtualcluster/pkg/resource-syncer/controller/controller.go]: Do you permanently lose one reconciler thread if Reconcile func returns false?

Comment (adohe) [reply incubator/virtualcluster/pkg/resource-syncer/controller/controller.go]: `go wait.Until(func() {
    for c.processNextWorkItem() {
    }
}, c.JitterPeriod, stop)`
IIUC this just means the for loop will break, the reconciler thread will not return, wait.Until will call worker func again after JitterPeriod.

Comment (Fei-Guo) [reply incubator/virtualcluster/pkg/resource-syncer/controller/controller.go]: Got you

**Pull request merged**